.github/workflows/pip_compile.yml

```
@@ -46,6 +46,7 @@ jobs:
         echo "ARCH=$(uname -m)" >> "$GITHUB_ENV"
     - name: install
       run: |
+        pip install --upgrade
```

```suggestion
        pip install --upgrade "pip<24.1"
```
as we discussed, you need to restore this otherwise it will continue failing